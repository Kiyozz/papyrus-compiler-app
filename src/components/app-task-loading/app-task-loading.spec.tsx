import React from 'react'
import { shallow } from 'enzyme'
import { CSSTransition } from 'react-transition-group'
import AppTaskLoading from './app-task-loading'

describe('<AppTaskLoading>', () => {
  it('renders correctly', () => {
    const component = (
      <AppTaskLoading loading={true} />
    )

    expect(component).toBeTruthy()
  })

  it('matches snapshot when not loading', () => {
    const component = shallow((
      <AppTaskLoading loading={false} />
    ))

    expect(component).toMatchSnapshot()
    expect(component.find(CSSTransition).prop('in')).toBe(false)
  })

  it('matches snapshot when loading', () => {
    const component = shallow((
      <AppTaskLoading loading={true} />
    ))

    expect(component).toMatchSnapshot()
    expect(component.find(CSSTransition).prop('in')).toBe(true)
  })
})
