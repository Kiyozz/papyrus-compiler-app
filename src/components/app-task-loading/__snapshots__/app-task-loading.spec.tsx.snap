// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppTaskLoading> matches snapshot when loading 1`] = `
<CSSTransition
  classNames="app-fade"
  in={true}
  mountOnEnter={true}
  timeout={300}
  unmountOnExit={true}
>
  <div
    className="app-task-loading"
  >
    <span
      className="app-task-loading-spin"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="circle-notch"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={true}
        symbol={false}
        title=""
        transform={null}
      />
    </span>
  </div>
</CSSTransition>
`;

exports[`<AppTaskLoading> matches snapshot when not loading 1`] = `
<CSSTransition
  classNames="app-fade"
  in={false}
  mountOnEnter={true}
  timeout={300}
  unmountOnExit={true}
>
  <div
    className="app-task-loading"
  >
    <span
      className="app-task-loading-spin"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="circle-notch"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={true}
        symbol={false}
        title=""
        transform={null}
      />
    </span>
  </div>
</CSSTransition>
`;
